src/php/ext/grpc/call.c

```diff
@@ -483,12 +488,28 @@ PHP_METHOD(Call, startBatch) {
       PHP_GRPC_DELREF(array);
       break;
     case GRPC_OP_RECV_MESSAGE:
+#if PHP_MAJOR_VERSION < 7
       byte_buffer_to_string(message, &message_str, &message_len);
+#else
+      zmessage = byte_buffer_to_zend_string(message);
```

Comment:
So is this the place where the "1 memcpy instead of 3" comes in? That you added this new `byte_buffer_to_zend_string` method to replace `byte_buffer_to_string` method?

Reply:
it's 2 to 1. the third one is in L510 using add_property_zval, instead of php_grpc_add_property_stringl